quartz/plugins/emitters/contentIndex.ts

```diff
@@ -116,50 +121,67 @@ export const ContentIndex: QuartzEmitterPlugin<Partial<Options>> = (opts) => {
     async emit(ctx, content, _resources) {
       const cfg = ctx.cfg.configuration
       const emitted: FilePath[] = []
-      const linkIndex: ContentIndex = new Map()
-      for (const [tree, file] of content) {
-        const slug = file.data.slug!
-        const date = getDate(ctx.cfg.configuration, file.data) ?? new Date()
-        if (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) {
-          linkIndex.set(slug, {
-            title: file.data.frontmatter?.title!,
-            links: file.data.links ?? [],
-            tags: file.data.frontmatter?.tags ?? [],
-            content: file.data.text ?? "",
-            richContent: opts?.rssFullHtml
-              ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
-              : undefined,
-            date: date,
-            description: file.data.description ?? "",
-          })
+      const feedIndices: Map<String, ContentIndex> = new Map()
+
+      // bfahrenfort: ts can't see the expansion of opts above that guarantees a non-null feedDirectories
+      const directories =
+        opts?.feedDirectories == null ? defaultOptions.feedDirectories : opts.feedDirectories
+
+      for (const feed of directories) {
+        const linkIndex: ContentIndex = new Map()
+        for (const [tree, file] of content) {
+          const slug = file.data.slug!
+
+          const date = getDate(ctx.cfg.configuration, file.data) ?? new Date()
+          if (
+            (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) &&
+            (slug.startsWith(feed) || feed == "index")
+          ) {
+            linkIndex.set(slug, {
+              title: file.data.frontmatter?.title!,
+              links: file.data.links ?? [],
+              tags: file.data.frontmatter?.tags ?? [],
+              content: file.data.text ?? "",
+              richContent: opts?.rssFullHtml
+                ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
+                : undefined,
+              date: date,
+              description: file.data.description ?? "",
+            })
+          }
         }
+        feedIndices.set(feed, linkIndex)
       }
 
       if (opts?.enableSiteMap) {
         emitted.push(
           await write({
             ctx,
-            content: generateSiteMap(cfg, linkIndex),
+            // bfahrenfort: "index" is guaranteed non-null
+            // see directories instantiation and feedIndices.set iterating over directories
+            content: generateSiteMap(cfg, feedIndices.get("index")!),
```

Comment:
```suggestion
            content: generateSiteMap(cfg, feedIndices.get("index")!),
```
ugh this seems brittle.